pkg/autodiscovery/listeners/docker.go

```diff
@@ -182,6 +182,11 @@ func (l *DockerListener) processEvent(e *docker.ContainerEvent) {
 	if found {
 		if e.Action == "die" {
 			l.removeService(cID)
+		} else if e.Action == "start" {
```

Comment:
small nit: this `if {} else if {} else{} can be replaced by a "switch case"

Reply:
yeah true - I just use switch cases when I want to suggest that there can be more use cases I have not considered or to be added in the future. Whereas here we should not have more. If that makes sense